src/main/java/com/endava/internship/warmup/domain/service/ArrayProcessorWithForLoops.java

```diff
@@ -1,153 +1,327 @@
      * @throws IllegalArgumentException when any array dimensions are not appropriate for matrix multiplication
      */
     @Override
     public void validateForMatrixMultiplication(int[][] leftMatrix, int[][] rightMatrix) throws NullPointerException, IllegalArgumentException {
-        throw new UnsupportedOperationException("Method not implemented yet");
+        try {
+            matrixMultiplication(leftMatrix, rightMatrix);
+        } catch (ArrayIndexOutOfBoundsException illegalArgumentException) {
+            throw new IllegalArgumentException("arr dim");
+        }
     }
 
     /**
      * Perform the matrix multiplication as described in previous example Please review the matrix multiplication
      * https://www.mathsisfun.com/algebra/matrix-multiplying.html
-     * @param leftMatrix the left matrix represented by array indexes [row][column]
+     *
+     * @param leftMatrix  the left matrix represented by array indexes [row][column]
      * @param rightMatrix the right matrix represented by array indexes [row][column]
-     * @throws NullPointerException when any of the inputs are null. (arrays, rows and columns)
+     * @throws NullPointerException     when any of the inputs are null. (arrays, rows and columns)
      * @throws IllegalArgumentException when any array dimensions are not appropriate for matrix multiplication
      */
     @Override
     public int[][] matrixMultiplication(final int[][] leftMatrix, final int[][] rightMatrix) throws NullPointerException, IllegalArgumentException {
-        throw new UnsupportedOperationException("Method not implemented yet");
+        int[][] mult = new int[leftMatrix.length][rightMatrix[0].length];
+        if (leftMatrix == null && rightMatrix == null) {
+            throw new IllegalArgumentException("gggg");
+        }
+        if (leftMatrix.length != rightMatrix[0].length) {
+            throw new IllegalArgumentException("not equal statement");
+        }
+        try {
+            for (int i = 0; i < leftMatrix.length; i++) {
+                for (int j = 0; j < rightMatrix[0].length; j++) {
+                    for (int k = 0; k < leftMatrix[0].length; k++) {
+                        mult[i][j] += leftMatrix[i][k] * rightMatrix[k][j];
+                    }
+                }
+            }
+        } catch (NullPointerException nullPointerException) {
+            nullPointerException.printStackTrace();
+        }
+        return mult;
     }
 
     /**
      * Return only distinct values in an array.
+     *
      * @param input non-null immutable array of ints.
      */
     @Override
     public int[] distinct(final int[] input) {
-        throw new UnsupportedOperationException("Method not implemented yet");
+        int[] disArr = new int[input.length];
+        for (int i = 0; i < disArr.length; i++) {
+            disArr[i] = input[i];
+        }
+        int[] dis_final = Arrays.stream(disArr).distinct().toArray();
```

Comment:
please try do this without using stream().distinct()

Reply:
can i use HashSet instead?

Reply:
no, in this task you must do everything using plain java loops

Reply:
Done